ncbi_genome_download/core.py

```diff
@@ -227,28 +227,33 @@ def download(**kwargs):
             metadata_table.write(get_table_header())
 
     # Actual logic
-    try:
-        download_jobs = []
-        for group in groups:
-            download_jobs.extend(
-                _download(section, group, uri, output, file_format, assembly_level, genus,
-                          species_taxid, taxid, human_readable, refseq_category, table))
 
-        pool = Pool(processes=parallel)
-        jobs = pool.map_async(worker, download_jobs)
-        try:
-            # 0xFFFF is just "a really long time"
-            jobs.get(0xFFFF)
-        except KeyboardInterrupt:  # pragma: no cover
-            # TODO: Actually test this once I figure out how to do this in py.test
-            logging.error("Interrupted by user")
-            return 1
+    def download_jobs_groups():
```

Comment:
Is there a particular reason you're switching to a generator here?

Reply:
I feel like the generator fits a bit better with the single threaded serially processing= mode (better than lists of jobs).